### PR TITLE
Update finagle lesson to follow latest finagle and style

### PR DESCRIPTION
**Pull request by stephbian**
@matteobanerjee 

Note: I got a little lazy towards the end and removed outdated information about configuring ClientBuilders without replacing it with how to configure the new server/client interface. This is largely because I'm not quite sure how to do the same kind of configuration in the new API.

### CODE REVIEW

File: web/finagle.textile [hunk 1]
```diff
@@ -57,39 +56,39 @@ future onSuccess { reply => // when the future gets "filled", use its value
 You can play with Futures in the REPL. This is a bad way to learn how you will use them in real code, but can help with understanding the API. When you use the REPL, Promise is a handy class. It's a concrete subclass of the abstract Future class. You can use it to create a Future that has no value yet.
 
 <pre>
-scala> import com.twitter.util.{Future,Promise}
-import com.twitter.util.{Future, Promise}
+scala> import com.twitter.util.{Future,Promise,Await}
+import com.twitter.util.{Future, Promise,Await}
```
Comment (matteosb) [inline web/finagle.textile]: rm space

File: web/finagle.textile [hunk 2]
```diff
@@ -482,26 +491,21 @@ h2(#client). Client
...
-// Don't worry, we discuss this magic "ClientBuilder" later
-val client: Service[HttpRequest, HttpResponse] = ClientBuilder()
-  .codec(Http())
-  .hosts("twitter.com:80") // If >1 host, client does simple load-balancing
-  .hostConnectionLimit(1)
-  .build()
+val client: Service[Request, Response] = Http.client.newService("localhost:80")
 
-val req = new DefaultHttpRequest(HttpVersion.HTTP_1_1, HttpMethod.GET, "/")
+val req = RequestBuilder().url("http://localhost:80/").buildGet()
```
Comment (matteosb) [inline web/finagle.textile]: why not just:

``` scala
val req = Request("/")
// OR:
val req = Request(Method.Get, "/")
```

**Comment by matteosb** [comment]
Don't worry about `configured` for server and client (remove them actually), I believe one of the reasons for the deprecation of Server/ClientBuilder was actually to make clients easier to build with sensible defaults. `Thrift.server.serveIface(...)` and `Thrift.client.newServiceIface(...)` are all we really need to show in a demo like this.

File: web/finagle.textile (hunk 1 above)
Comment (stephbian) [inline web/finagle.textile]: done

File: web/finagle.textile (hunk 2 above)
Comment (stephbian) [inline web/finagle.textile]: done

**Pull request merged**